app/controllers/appliance_comments_controller.rb

```diff
@@ -5,7 +5,7 @@ class ApplianceCommentsController < ApplicationController
   # GET /appliance_comments.json
   def index
     @appliance_comments = Appliance.includes(:appliance_comments).all
-                                   .map{ |appliance| appliance.comments.last }
+                                   .filter_map{ |appliance| appliance.comments.last if !appliance.comments.last.nil? }
                                    .sort_by(&:updated_at).reverse
   end
```

Comment:
It seems, if the appliance didn't have any comments, the last method would give back `nil`, with which `updated_at` in `sort_by` would always break the page from loading.

Reply:
The filter_map is a clever solution for getting only the existing comments.